### PR TITLE
Add more specific class names to notification divs

### PR DESCRIPTION
**Pull request by chris-martin**
I found myself wanting to add some custom CSS to fav/boost notifications, but there weren't any CSS classes to distinguish them. This diff just adds some classes. It shouldn't directly affect how anything renders in the default style.

### CODE REVIEW

**Comment by yiskah** [comment]
Unsure if we should merge this when the classes don't call anything in the official unmodified code. Seems messy.

**Comment by ineffyble** [comment]
If the class names are similar to current ones, and don't clash, then adding them as "placeholder" classes doesn't feel inherently bad.

File: app/assets/javascripts/components/features/notifications/components/notification.jsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ const Notification = React.createClass({
 
   renderFollow (account, link) {
     return (
-      <div className='notification'>
+      <div className='notification notification--follow'>
```
Comment (Gargron) [inline app/assets/javascripts/components/features/notifications/components/notification.jsx]: When it's a namespaced (nested) element: `parent__child`. When it's a modifier: `parent-modifier`. I see nothing wrong with this PR in general but the format of these needs to be fixed.

**Review by Gargron — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/components/features/notifications/components/notification.jsx (hunk 1 above)
Comment (chris-martin) [inline app/assets/javascripts/components/features/notifications/components/notification.jsx]: Oh sorry, when I saw the class names with `__` I assumed you were following [BEM](http://getbem.com/introduction/). Will update.

**Comment by chris-martin** [comment]
@Gargron Updated with requested changes.

**Review by yiskah — APPROVED** [review]
These look to be the requested changes

**Review by Gargron — APPROVED** [review]

**Pull request merged**